plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/TrinoToMySqlWriteChecker.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mysql;
+
+import io.trino.spi.TrinoException;
+
+import java.time.Instant;
+
+import static io.trino.spi.StandardErrorCode.INVALID_ARGUMENTS;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class TrinoToMySqlWriteChecker<T>
+{
+    // The range for TIMESTAMP values is '1970-01-01 00:00:01.000000' to '2038-01-19 03:14:07.499999'
+    public static final TrinoToMySqlWriteChecker<Instant> TIMESTAMP = new TrinoToMySqlWriteChecker<>(
+            new TimestampWriteValueChecker(new Range<>(Instant.parse("1970-01-01T00:00:01.000000Z"), Instant.parse("2038-01-19T03:14:07.499999Z"))));
+
+    private final Checker<T> checker;
+
+    private TrinoToMySqlWriteChecker(Checker<T> checker)
+    {
+        this.checker = requireNonNull(checker, "checker is null");
+    }
+
+    public void validate(T value)
+    {
+        checker.validate(value);
+    }
+
+    private interface Checker<T>
+    {
+        void validate(T value);
+    }
+
+    private static class TimestampWriteValueChecker
+            implements Checker<Instant>
+    {
+        private final Range<Instant> range;
+
+        public TimestampWriteValueChecker(Range<Instant> range)
+        {
+            this.range = requireNonNull(range, "range is null");
+        }
+
+        @Override
+        public void validate(Instant value)
+        {
+            if (value.isBefore(range.getMin()) || value.isAfter(range.getMax())) {
```

Comment:
In other places in the project (like io.trino.plugin.jdbc.JdbcMetadataConfig), I see we seem to be using a validation framework, for instance jakarta.validation.constraints.Min. Is there a reason (performance?) we cannot use it here as well?